sqlglot/expressions.py

```diff
@@ -484,43 +484,78 @@ class Interval(Expression):
 # Functions
 class Func(Expression):
     token_type = TokenType.FUNC
+    ordered_arg_types = [('this', True)]
+    is_var_len_args = False
+
+    def __init__(self, **args):
+        self.arg_types = dict(self.ordered_arg_types)
+        super().__init__(**args)
+
+    @classmethod
+    def from_arg_list(cls, args):
+        args_num = len(args)
+
+        # If this function supporst variable length argument treat the last argument as such.
+        arg_types = cls.ordered_arg_types[:-1] if cls.is_var_len_args else cls.ordered_arg_types
+
+        args_dict = {}
+        arg_idx = 0
+        for arg_key, _ in arg_types:
```

Comment:
enumerate(arg_types) instead of manual arg_index?

Reply:
Good point 👍 I forgot the fact that in Python whatever is defined in the `for`'s scope is available outside the block.

Reply:
Provided I use `arg_idx` outside the `for` block, `enumerate` doesn't really do the right thing. Example:
```
>>> idx = 0
>>> for idx in range(0, 1):
...     pass
...
>>> idx
0
```
My expectation is that the `arg_idx` should point to the next index after the last successful iteration, not the index of the last successful iteration itself.